fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -179,6 +183,48 @@ private IndexingByIndex getIndexerByIndex() {
         if (ex == null) {
             return AsyncUtil.DONE;
         }
+        IndexBuildProto.IndexBuildIndexingStamp conflictingIndexingTypeStamp = common.popConflictingIndexingTypeStamp();
+        if (conflictingIndexingTypeStamp != null) {
+            // Here: an indexing type stamp already exists, with a different type. Few precondition cases are handled
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info(KeyValueLogMessage.build("conflicting indexing type stamp",
+                        LogMessageKeys.INDEX_NAME, common.getIndex().getName(),
+                        LogMessageKeys.INDEX_VERSION, common.getIndex().getLastModifiedVersion(),
+                        LogMessageKeys.INDEXER_ID, common.getUuid(),
+                        LogMessageKeys.INDEX_STATE_PRECONDITION, common.getIndexStatePrecondition(),
+                        LogMessageKeys.ACTUAL_TYPE, conflictingIndexingTypeStamp
+                ).toString());
+            }
+            if (common.getIndexStatePrecondition() == IndexStatePrecondition.BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY) {
+                indexer = null;
+                // now, match the policy to the previous run
+                IndexBuildProto.IndexBuildIndexingStamp.Method method = conflictingIndexingTypeStamp.getMethod();
+                if ( method == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
+                    fallbackToRecordsScan = true;
+                    return fallbackFunction.get();
+                }
+                if (method == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_INDEX) {
+                    Object sourceIndexSubspaceKey = decodeSubspaceKey(conflictingIndexingTypeStamp.getSourceIndexSubspaceKey());
+                    indexFromIndexPolicy = IndexFromIndexPolicy.newBuilder()
+                            .setSourceIndex(sourceIndexSubspaceKey.toString())
```

Comment:
Hm, the index subspace key doesn't have to be the index name, and so in general, this is incorrect. I guess at the moment, this doesn't cause problems because if a source index subspace key is set, the source index (name) is ignored, but it would seem better to let the builder set the index subspace key, and then leave the source index blank.

Or in other words, it should only set the source index to the source index name, not to something bogus.

Also, is there a test that exercises this code path (i.e., resumes a build with a different index from the original one provided)?

Reply:
It was meant to be used for debugging only, but I've eliminated it with previous changes (setting the subspace key at the builder).
`testIndexFromIndexPersistentContinuePreviousByIndex` should be covering this code.